### ISSUE TITLE
[Evals] The displayed errors contradict each other.

### ISSUE THREAD

**Issue by ze1enskam** [Arize-ai/phoenix#3038]
The “HallucinationEvaluator(eval_model)” and “QAEvaluator(eval_model)” codes are highlighted as an error in the “Define evaluators block” and the error message “Argument missing for parameter "template" is displayed. 
<img width="601" alt="Screenshot 2024-04-29 at 16 03 54" src="https://github.com/Arize-ai/phoenix/assets/119803899/6a11fa8c-af9f-4577-9e11-2d99d3b4f3e6">

If the template is added to the arguments, e.g.:
```from tempfile import template
from phoenix.experimental.evals import HallucinationEvaluator, QAEvaluator
from phoenix.experimental.evals import HALLUCINATION_PROMPT_TEMPLATE, QA_PROMPT_TEMPLATE
from phoenix.experimental.evals import run_evals
from phoenix.session.evaluation import get_qa_with_reference



# Define your evaluators
hallucination_evaluator = HallucinationEvaluator(eval_model,HALLUCINATION_PROMPT_TEMPLATE)
qa_evaluator = QAEvaluator(eval_model, QA_PROMPT_TEMPLATE)

# Log the evaluations
from phoenix.trace import SpanEvaluations

(client).log_evaluations(
    SpanEvaluations(eval_name="Hallucination", dataframe=hallucination_eval_df),
    SpanEvaluations(eval_name="QA Correctness", dataframe=qa_eval_df)
) 
```
This resolves the “Argument missing” issue, but another “TypeError” error is displayed:
```

TypeError                                 Traceback (most recent call last)
[/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb](https://file+.vscode-resource.vscode-cdn.net/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb) Cell 7 line 1
      [5](vscode-notebook-cell:/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb#W5sZmlsZQ%3D%3D?line=4) from phoenix.session.evaluation import get_qa_with_reference
      [9](vscode-notebook-cell:/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb#W5sZmlsZQ%3D%3D?line=8) # Define your evaluators
---> [10](vscode-notebook-cell:/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb#W5sZmlsZQ%3D%3D?line=9) hallucination_evaluator = HallucinationEvaluator(eval_model,HALLUCINATION_PROMPT_TEMPLATE)
     [11](vscode-notebook-cell:/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb#W5sZmlsZQ%3D%3D?line=10) qa_evaluator = QAEvaluator(eval_model)
     [14](vscode-notebook-cell:/Users/marynazelenska/Documents/GitHub/qira-chatbot/notebooks/phoenix/testeval.ipynb#W5sZmlsZQ%3D%3D?line=13) queries_df = get_qa_with_reference(client)

TypeError: _create_llm_evaluator_subclass.<locals>.__init__() takes 2 positional arguments but 3 were given
```
The reason for this behavior might be the following code in the “evaluators.py” doc:
The probable reason for the error message “Argument missing for parameter "template", is that the LLMEvaluator class includes 3 arguments:

``` class LLMEvaluator:
    """
    Leverages an LLM to evaluate individual records.
    """

    def __init__(
        self,
        model: BaseEvalModel,
        template: ClassificationTemplate,
    ) -> None:
        """Initializer for LLMEvaluator.

        Args:
            model (BaseEvalModel): The LLM model to use for evaluation.
            template (ClassificationTemplate): The evaluation template.
        """
        self._model = model
        self._template = template
```
Further, in the _create_llm_evaluator_subclass another init function is defined (with 2 arguments).
The type() function is used to create the subclass with the given class_name dynamically, inherits from LLMEvaluator, and includes the modified __init__ function. The return type (Type[LLMEvaluator]) does not match with the actual returned class (modified __init__ method). What could occur “TypeError”.

```
def _create_llm_evaluator_subclass(
    class_name: str, template: ClassificationTemplate, docstring: str
) -> Type[LLMEvaluator]:
    """A factory method that dynamically creates subclasses of LLMEvaluator.

    Args:
        class_name (str): Name of the class to be created (should match the name
        of the assignment variable).

        template (ClassificationTemplate): The classification template to use
        for evaluation.

        docstring (str): The docstring that will be attached to the subclass.

    Returns:
        Type[LLMEvaluator]: The dynamically created subclass.
    """

    def __init__(self: LLMEvaluator, model: BaseEvalModel) -> None:
        LLMEvaluator.__init__(self, model, template)

    __init__.__doc__ = f"""
        Initializer for {class_name}.

        Args:
            model (BaseEvalModel): The LLM model to use for evaluation."""

    docstring += f" Outputs railed classes {', '.join(template.rails)}."
    docstring += "\n\nThe template used for evaluation (without explanation) is:\n\n"
    docstring += indent(template.template, 2 * _TAB)

    return type(class_name, (LLMEvaluator,), {"__init__": __init__, "__doc__": docstring})
```

Environment
 - OS: [MacOS]
 - Notebook Runtime [Jupyter]
 - Browser [Chrome]
 - Version [124.0.6367.93]

**Comment by mikeldking** [comment]
@ze1enskam Thanks for the detailed report! We are actually actively moving off experimental. I'm wondering if that is the source of conflicting types. Will look into this further if you continue to face issues. Let us know!

**Comment by axiomofjoy** [comment]
@ze1enskam Thanks for reporting this issue and for the detailed analysis. I believe this is a duplicate of https://github.com/Arize-ai/phoenix/issues/2638. I tried to get too fancy in the way we define these evaluators. Closing this issue as it is a duplicate, we will fix.

**Comment by axiomofjoy** [comment]
This issue is resolved in `arize-phoenix-evals==0.8.2`.
